core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -56,16 +56,18 @@
 
         this.properties.putAll(classpathProperties);
         this.properties.putAll(environmentProperties);
+        properties.keySet()
+            .forEach(key -> properties.replace(key, properties.getProperty(String.valueOf(key)).trim()));
     }
 
     @Deprecated
     public String getAmbassadorContainerImage() {
-        return (String) properties.getOrDefault("ambassador.container.image", "richnorth/ambassador:latest");
+        return String.valueOf(properties.getOrDefault("ambassador.container.image", "richnorth/ambassador:latest"));
```

Comment:
do we still need `valueOf` here? Thinking of `"null"` String being returned instead of `null` if it is `null` for some reason, that will make it harder to debug

Reply:
As I understand correctly `gotOrDefault` will either return value in property or in case of null value it will return that second value (in this case `richnorth/ambassador:latest`), maybe I am missing something, but `"null"` String can be returned only when `ambassador.container.image=null` in properties file